system-tests/supertest/src/e2e/saveList/saveList.test.js

```diff
@@ -0,0 +1,67 @@
+import assert from 'assert';
+import request from 'supertest';
+
+describe('saving a todo list', () => {
+  it('responds with "Bad Request" when list is empty', (done) => {
+    request('https://bff-fg5blhx72q-ey.a.run.app')
+      .post('/toDoLists')
+      .send({
+        name: 'some list',
+        todoItem: [],
+      })
+      .set('Accept', 'application/json')
+      .expect(400)
+      .then((response) => {
+        assert(response.body.error === "EMPTY_LIST",'list is empty error');
+        //assert(response.body.services.find((it) => it.name === 'todo').serving, 'service todo ok');
+        done();
+    });
+  });
+
+    it('responds ok when list is created successfully', (done) => {
+        request('https://bff-fg5blhx72q-ey.a.run.app')
+            .post('/toDoLists')
+            .send({
+                name: 'some list',
+                todoItem: [
+                    {
+                        description: 'feed the cat',
+                    },
+                ],
+            })
+            .set('Accept', 'application/json')
+            //.expect('Content-Type', /json/)
+            .expect(200)
+        done();
+    });
+});
+
+it('responds not ok when list already exists', (done) => {
```

Comment:
@ufarooq82 nice, thanks for tackling that.
what we need to do here is set up the test environment, bring it to a state where the list `name already taken` exists already. This test script should take care of it's own preconditions, makes it easier, all info in one place.

So what we need to do is either:
- check for the existence by calling `GET /toDoLists/name already taken` and if not exists, create one
- or always create on with a random name like `const name = "list" + Date.now()` and then use that in the post request body

does that make sense to you?